### ISSUE TITLE
Auto Import only on key press

### ISSUE THREAD

**Issue by spinelsun** [microsoft/pylance-release#178]
Hi,

I would like to have the ability to choose when the auto-import actually imports modules, functions, etc.
Currently, when I press Tab or Enter on IntelliSense completion suggestion the auto-import always imports the suggestion even if it is not necessary.
For example, when writing tests and using fixtures there is no need to import fixture from conftest.py or any other global module.
Another example, if I define a const variable name `EXAMPLE` in one module when I try to create `EXAMPLE` in another module auto-import will import the first one

I think that adding a keystroke to auto-import will be the best solution. Maybe when IntelliSense suggests something enter will take this suggestion without auto-import and tab will auto-import.

Thanks

**Comment by jakebailey** [comment]
Note that this would need to be a VS Code change and LSP change; neither offer this sort of thing. Completions are a request purely dependent on the file state and the location. A server is typically expected to return the same thing each time, and the client is free to cache requests and not ask the server again.

**Comment by yzhang-gh** [comment]
I second this (it will be better if we make the auto-import operation more perceivable).

The story is: I found there were unexpected auto-imports several times (_they were not noticeable until I ran the code_) so I disabled this feature. Today I re-enabled it as it is indeed a nice feature. But soon I again got an unexpected auto-import. (I wanted to claim a variable named `cluster` but `sklearn.cluster` was auto-imported without awareness... 😑) It's so frustrating.

I think the problem is the mismatch between _user expectation_ and _the auto-import behaviors in practice_. In short, users cannot notice unexpected auto-imports **in time**. (A naive solution I could think of is to show a status message (last for 2 seconds) whenever we have an auto-import.)
